webplugin/js/app/km-widget-custom-events.js

```diff
@@ -0,0 +1,196 @@
+/**
+ // Kommunicate custom events available for user for the specific event in chat widget.
+ // resp - JSON object containing eventAction and other property exposed to user.
+ // defination of function can be overwrite by user through subscribeToEvents function.
+ */
+var kmWidgetEvents = {
+    sendEventToGoogleAnalytics: function (
+        eventCateogry,
+        eventAction,
+        eventLabel,
+        eventValue
+    ) {
+        var trackingID = applozic._globals.gaTrackingID;
+        if (trackingID && typeof window.top.ga !== 'undefined') {
+            window.top.ga('create', trackingID.toString(), 'auto');
+            window.top.ga('send', {
+                hitType: 'event',
+                eventCategory: eventCateogry,
+                eventAction: eventAction,
+                eventLabel: eventLabel,
+                eventValue: eventValue,
+            });
+        }
+    },
+    eventTracking: function (eventObject, customLabel, customValue) {
+        // Any other analytics tool related code can be add here no need to paste it in every function
+        if (kommunicateCommons.isObject(eventObject)) {
+            if (customLabel) {
+                eventObject.data.eventLabel = customLabel;
+            }
+            if (customValue) {
+                eventObject.data.eventValue = customValue;
+            }
+            applozic._globals.gaTrackingID &&
+                kmWidgetEvents.sendEventToGoogleAnalytics(
```

Comment:
we are accessing this object in the definition itself. Does it work? Not sure about this, please verify once.

Reply:
pls check this fiddle- https://jsfiddle.net/csguj25o/